File: app/src/main/java/com/unbounds/trakt/progress/WatchedProgressWrapper.java
package com.unbounds.trakt.progress;

import com.unbounds.trakt.api.model.Show;
import com.unbounds.trakt.api.model.response.WatchedProgress;

/**
 * Created by maclir on 2015-11-22.
 */
class WatchedProgressWrapper {
    private final Show mShow;
    private WatchedProgress mWatchedProgress;
    private boolean selected;

    WatchedProgressWrapper(final WatchedProgress watchedProgress, final Show show) {
        mWatchedProgress = watchedProgress;
        mShow = show;
    }

    Show getShow() {
        return mShow;
    }

    WatchedProgress getWatchedProgress() {
        return mWatchedProgress;
    }

    boolean isSelected() {
        return selected;
    }

    WatchedProgressWrapper setWatchedProgress(final WatchedProgress watchedProgress) {
        mWatchedProgress = watchedProgress;
        return this;
    }

    void setSelected(final boolean selected) {
        this.selected = selected;
    }
}

File: app/src/main/java/com/unbounds/trakt/view/progress/WatchedProgressWrapper.kt
package com.unbounds.trakt.view.progress

import com.unbounds.trakt.service.api.model.trakt.Show
import com.unbounds.trakt.service.api.model.trakt.response.WatchedProgress

/**
 * Created by maclir on 2015-11-22.
 */
data class WatchedProgressWrapper(
        val show: Show,
        var watchedProgress: WatchedProgress? = null,
        var isSelected: Boolean = false,
)
